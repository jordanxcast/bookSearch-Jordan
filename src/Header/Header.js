import React, { Component } from 'react';
import SearchBar from '../SearchBar/SearchBar';
import Filter from '../Filter/Filter';

class Header extends Component {
  render(){
    return(
      <header>
        <div className='title' >
          <h1>Google Book Search</h1>
        </div>
        <SearchBar searchTerm={this.props.changeSearchTerm}/>
        <Filter 
        changePrintType={this.props.changePrintType}
        changeBookType={this.props.changeBookType}
        changeViewability={this.props.changeViewability}/>
      </header>
    );
  }
}

export default Header;