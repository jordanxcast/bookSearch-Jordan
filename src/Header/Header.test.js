import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe('Header Component', ()=>{
  it('renders with empty div', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<Header />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders header as is', ()=>{
    const wrapper = shallow(<Header />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
});