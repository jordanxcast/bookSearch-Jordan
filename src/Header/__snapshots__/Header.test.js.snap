// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders header as is 1`] = `
<header>
  <div
    className="title"
  >
    <h1>
      Google Book Search
    </h1>
  </div>
  <SearchBar />
  <Filter />
</header>
`;
